app/services/job_runner/job_configuration.rb

```diff
@@ -0,0 +1,30 @@
+module JobRunner
+  class JobConfiguration
+    attr_reader :name
+    attr_reader :interval
+    attr_reader :timeout
+    attr_reader :callback
+
+    # :reek:ControlParameter
+    def initialize(name:, interval:, timeout: nil, callback:)
+      @name = name
+      @interval = interval
+      @timeout = timeout || interval
+      @callback = callback
+    end
+
+    def to_s
+      "JobConfiguration #{name.inspect}"
+    end
+
+    def run_if_needed
+      Rails.logger.debug("#{self}: Running job")
+      LockReferee.new(self).acquire_lock_and_run_callback_if_needed
+    end
+
+    def clean_up_timeouts
```

Comment:
This only marks as timed out. We probably want a mechanism to actually terminate long running jobs.

Reply:
Oh right, because even after the code runs to update the DB table, I don't see code to kill any jobs.,

Reply:
Repeating our conversation IRL: I think it's OK that we don't have a way to terminate long running jobs. In practice we don't expect that to happen unless something has gone horribly wrong (like the database being unavailable). And the type of cleanup needed is likely to depend on the specific nature of the jobs in question.

If it turns out that we do have jobs timing out all the time, then we may want to add some more robust kill-switch timer, such as by using Timeout::Timeout or wrapping the entire process in a timeout.